src/Symfony/Component/Validator/Constraints/Composite.php

```diff
@@ -135,8 +148,17 @@ abstract protected function getCompositeOption(): string;
      */
     public function getNestedConstraints(): array
     {
-        /* @var Constraint[] $nestedConstraints */
-        return $this->{$this->getCompositeOption()};
+        $compositeOptions = $this->getCompositeOption();
+        if (!\is_array($compositeOptions)) {
+            $compositeOptions = [$compositeOptions];
+        }
+
+        $constraints = [];
+        foreach ($compositeOptions as $compositeOption) {
+            $constraints = array_merge($constraints, (array) $this->$compositeOption);
```

Comment:
let's use this pattern everywhere possible to make the code simpler:
```suggestion
        $constraints = [];
        foreach ((array) $this->getCompositeOption() as $option) {
            $constraints = array_merge($constraints, (array) $this->$option);
```

Reply:
Here you go